### PR TITLE
Bash function to setup Helm

### PR DESCRIPTION
**Pull request by alpeb**
This introduces the `install_helm` bash function that installs the
`helm` binary under `/target`, and optionally Tiller in the current
cluster.

To be consumed by the Helm helper scripts for development, and by
integration tests.

Ref #3143

Signed-off-by: Alejandro Pedraza <alejandro@buoyant.io>

### CODE REVIEW

File: bin/_helm.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+set -eu
+
+# Checks if target/helm exists and it installs it if it doesn't.
+# Use `install_helm true` to also install Tiller.
+install_helm() {
+    bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+    rootdir="$( cd $bindir/.. && pwd )"
+
+    if [ ! -f $rootdir/target/helm ]; then
```
Comment (siggy) [inline bin/_helm.sh]: name the executable with the version number, similar to protoc, that way when we update the version in this script, it'll re-download.

File: bin/_helm.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        if [ "$(uname -s)" = "Darwin" ]; then
+            os=darwin
+        else
+            os=linux
```
Comment (siggy) [inline bin/_helm.sh]: maybe switch on arch as well?
```bash
if [ "$(uname -s)" = "Darwin" ]; then
    os=darwin
    arch=amd64
else
    os=linux
    case $(uname -m) in
        x86_64) arch="amd64" ;;
        arm)    dpkg --print-architecture | grep -q "arm64" && arch="arm64" || arch="arm" ;;
    esac
fi
```

File: bin/_helm.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -eu
+
+# Checks if target/helm exists and it installs it if it doesn't.
+# Use `install_helm true` to also install Tiller.
```
Comment (siggy) [inline bin/_helm.sh]: `install_helm` without params fails with `bash: $1: unbound variable`, just confirming that's expected.

Comment (alpeb) [reply bin/_helm.sh]: Good catch, I though I had that covered :-(

**Comment by alpeb** [comment]
@siggy thanks for the feedback; all points agreed upon and implemented :+1:

**Comment by ihcsim** [comment]
The new script looks good :+1:. I think we can remove the existing `bin/helm.sh` and update `travis.yaml` to call this script. Optionally, we can also tackle #3199 as part of this branch by updating `BUILD.md`?

File: bin/helm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        cd "$tmp"
+        curl -Lsf -o "./helm.tar.gz" "$helmcurl"
+        tar zf "./helm.tar.gz" -x "$targetdir"
+        chmod +x "$targetdir/helm"
+    )
+    mv "$tmp/$targetdir/helm" "$helmbin"
+    rm -rf "$tmp"
+fi
+
+$helmbin "$@"
```
Comment (siggy) [inline bin/helm]: nit: newline at end of file (my editor will automatically end up doing this eventually)

**Comment by alpeb** [comment]
I'm merging this one, and punting the dev workflow notes and pending .travis.yml work to #3199 and #3230

File: bin/helm (hunk 4 above)
Comment (alpeb) [reply bin/helm]: oops I forgot to push this one before merging. Will take care of it in #3223

**Pull request merged**